### DISCUSSION TITLE
Looking for help maintaining project

### DISCUSSION THREAD

**Discussion by toddbluhm** [toddbluhm/env-cmd#371] [Announcements]
Hello Community,

First I would like to say a big THANK YOU to the entire community that utilized and relies on this project. You are all the best and I am sorry that I have not been as active as I used to be.

It has been way too long since I have given the proper attention to this project and I sincerely apologize to the community. I have since moved on from software engineering and I am now in a new career field. I still have the desire to maintain my software skills and assist on this project, however, I have not been keeping up with the latest in the nodejs world and as such have fallen way behind.

I am not really looking to just give someone this project and walk away because I believe it is important to the community that the person they have placed their trust in for this project continues to be a part of this, however, I do believe it would be very helpful to have a few people on board to assist in keeping the project up to date when I am unable to do so and to help get this project moving forward again.

If you are interested in assisting in maintaining this project, please comment and let me know. Specifically, I would like to see how you are using the project, your ideas for updating/improving the project, how active you have been in other communities, and how much time you would like to spend on this project. This is a small and hopefully very focused project so I am not expecting vast amounts of time on it, just maintenance time.

Lastly, once a few people have been selected, my hope is to get everyone on an audio call to quickly discuss where this project is at, the direction it should go, and the expectations of everyone involved. Again, not expecting a huge time sink, but I think it is important that everyone be on the same page before new development and maintenance is started.

**Comment by k-yle** [comment]
I'd be happy to help out with maintenance and triaging issues. I don't have any grand plans for improving env-cmd since it already does everything I need it to :) although support for other file formats would be nice (e.g. #133), we wouldn't want to force new dependencies onto existing users. 

With that said, I'm mostly interested in fixing existing bugs, such as #339, #343, #238, and presumably other bugs that I haven't noticed myself.

**Comment by iamhosseindhv** [comment]
Hey @toddbluhm - Well done creating this project and hope you're enjoying your new career path.

I have used env-cmd for years at the companies I've worked at, primarily to load certain env variables during local dev, as usually env variables are loaded differently in the cloud environment. In personal projects, I have used this library to load env vars on the cloud too.

I have been maintaining [notistack](https://github.com/iamhosseindhv/notistack) since 2017 and have some experience maintaining open source projects.

If I was working on this project, my short-term goal would be to immediately make a patch release, so it doesn't appear that this library is unmaintained. The last stable release was made 4 years ago, and unfortunately that can be considered as a red flag and difficult to justify, when a dev wants to adapt this library at the company they work at. 

<img width="298" alt="Screenshot 2023-12-14 at 19 28 58" src="https://github.com/toddbluhm/env-cmd/assets/26996372/0f1fa088-b632-464f-a3ec-e93d7fb7add2">

More long term goal, would be to increase adaption of the library, by returning more human friendly errors. All these 10 issues are about the same problem and I have personally suffered from this bug (in terms of time wasted): https://github.com/toddbluhm/env-cmd/issues?q=is%3Aissue+is%3Aopen+failed+to+find

Cleaning up GitHub issues of the old/inactive threads, can allow us better see what users care about and want to see happening in this repo. Be assured, I will never keep adding features to this package because some random person suggests it. My goal is to make sure this package does the basic tasks of loading env vars in the most flexible and simple way possible.

A very simple documentation website can be quickly spun up which further demonstrates that this is a well maintained package. 

Things have changed since 4 years ago, first-class typescript support is a must for this library if it wants to stay relevant over time.

Launching this library on Product Hunt (the way [I did it](https://www.producthunt.com/products/notistack#notistack) for notistack) can also increase adaption / downloads.

Looking forward to hear from you

**Comment by toddbluhm** [comment]
@k-yle I took a look at your work and was impressed by it. I think if you are interested, I would be happy to add you as a maintainer and publisher. We can openly discuss any features or improvements for this project you would find helpful. Also, helping triage the current issues would be welcomed as well. 

@iamhosseindhv I also took a look at your work at `notistack` and while it was also impressive, you definitely put a better marketing spin on your work. This project is more a utility and I kind of like keeping it on the simpler side, less flashy. However, if you would like to provide PR's to help improve the project, I will certainly welcome them...especially with the more human-friendly errors suggestion!

I do appreciate both of you and welcome any help and support your willing to give.

FYI: I just committed a largish refactor to the project, more just updating the dependencies which led to converting the project over to `esm` modules and updating to use `typescript-eslint` over `ts-standard` and updating project for node `>=18`.

**Reply by k-yle** [reply]
Thanks @toddbluhm, yes I'm still interested in helping out :) I also just noticed your recent commits which drop support for node v10-16, so I assume you want to release all bugfixes & future changes as a new major version (v11)?

**Reply by toddbluhm** [reply]
Yes, that is correct. 

I had attempted a while ago to update `env-cmd` to one of those earlier node versions but ran into many import/require issues just do to the partial implementation of `esm` modules and many compatibility issues. This time around, `esm` is far enough along that I could just upgrade everything to `esm` without too much of an issue. The biggest hurdle and may be a problem in the future is doing mock/stubbing with `esm`. `esmock` is working well, but I suspect a future node version will break their implementation at some point.

If people need an issue fix on `v10` specifically we can either back-port the fix or try to release a patch. I do suspect though that few people are still using those old versions of node anymore.
